### PR TITLE
Add iterators

### PR DESCRIPTION
**Pull request by XuanYang-cn**
See also: #27606

### CODE REVIEW

File: internal/datanode/iterators/binlog_iterator.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+func (i *BinlogIterator) isDisposed() bool {
+	return i.disposed.Load()
+}
+
+// Disposed wait forever for the iterator to dispose
+func (i *BinlogIterator) WaitForDisposed() <-chan struct{} {
+	<-i.disposedCh
+	notifyCh := make(chan struct{})
+	close(notifyCh)
+	return notifyCh
```
Comment (congqixia) [inline internal/datanode/iterators/binlog_iterator.go:105]: why return a newly created closed channel here? this method already wait iterate disposal done

File: internal/datanode/iterators/deltalog_iterator.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+func (d *DeltalogIterator) isDisposed() bool {
+	return d.disposed.Load()
+}
+
+func (d *DeltalogIterator) WaitForDisposed() <-chan struct{} {
+	<-d.disposeCh
+	notifyCh := make(chan struct{})
+	close(notifyCh)
+	return notifyCh
```
Comment (congqixia) [inline internal/datanode/iterators/deltalog_iterator.go:82]: ditto

File: internal/datanode/iterators/binlog_iterator.go (hunk 1 above)
Comment (XuanYang-cn) [reply internal/datanode/iterators/binlog_iterator.go:105]: True

**Review by congqixia — APPROVED** [review]
/kind improvement
/lgtm

**Pull request merged**